util/Pos/Util/Util.hs

```diff
@@ -426,23 +427,28 @@ sleep n = liftIO (threadDelay (truncate (n * 10^(6::Int))))
 
 -- | 'tMeasure' with 'logDebug'.
 tMeasureLog :: (MonadIO m, WithLogger m) => Text -> m a -> m a
-tMeasureLog = tMeasure logDebug
+tMeasureLog label = fmap fst . tMeasure logDebug label
 
 -- | 'tMeasure' with 'putText'. For places you don't have
 -- 'WithLogger' constraint.
 tMeasureIO :: (MonadIO m) => Text -> m a -> m a
-tMeasureIO = tMeasure putText
+tMeasureIO label = fmap fst . tMeasure putText label
+
+timed :: (MonadIO m, WithLogger m) => Text -> m a -> m (a, Microsecond)
+timed = tMeasure logDebug
```

Comment:
I have generalised `tMeasure` to be able to return the time it took to execute the passed action.